### PR TITLE
Allow to extends constant variable(master)

### PR DESCRIPTION
**Pull request by sanikoyes**
this pr same to : https://github.com/godotengine/godot/pull/12506

### CODE REVIEW

File: modules/gdscript/gd_parser.cpp [hunk 1]
```diff
@@ -2963,18 +2963,37 @@ void GDParser::_parse_extends(ClassNode *p_class) {
...
-			return;
-		}
+		switch (tokenizer->get_token()) {
+
+			case GDTokenizer::TK_IDENTIFIER: {
+
+					StringName identifier = tokenizer->get_token_identifier();
+					p_class->extends_class.push_back(identifier);
+				}
+				break;
```
Comment (akien-mga) [inline modules/gdscript/gd_parser.cpp]: It looks like clang-format doesn't like the indentation style you used: https://travis-ci.org/godotengine/godot/jobs/294820735

**Comment by akien-mga** [comment]
Pre-approved in PR review meeting, will merge tomorrow once 3.0 alpha2 is released, to avoid potential regressions.

File: modules/gdscript/gd_compiler.cpp [hunk 2]
```diff
@@ -1685,21 +1685,44 @@ Error GDCompiler::_parse_class(GDScript *p_script, GDScript *p_owner, const GDPa
...
 					String subclass = p_class->extends_class[i];
 
+					ident += ("." + subclass);
+
 					if (base_class->subclasses.has(subclass)) {
 
 						base_class = base_class->subclasses[subclass];
+					} else if (base_class->constants.has(subclass)) {
+
+						Ref<GDScript> base = base_class->constants[subclass];
```
Comment (Marqin) [inline modules/gdscript/gd_compiler.cpp]: You are overloading your outer scope variable `String base` here and I don't think it's a good practice. Try using some different name, for example `Ref<GDScript> new_base_class` would be better imho.

**Review by Marqin — CHANGES_REQUESTED** [review]

**Comment by akien-mga** [comment]
Sorry for the delay - I've done a refactoring of the gdscript module in #12969 so there are now merge conflicts to resolve. As the file names changed and all `GD*` classes are now prefixed with `GDScript`, the faster solution might be to reimplement your changes on top of a clean branch and force push it to override this one.

**Comment by akien-mga** [comment]
Merged manually as fb801d496 to fix the merge conflicts (and take @Marqin's review into account).

**Pull request closed without merging**